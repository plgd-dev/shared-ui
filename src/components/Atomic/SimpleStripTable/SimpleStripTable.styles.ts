import { css } from '@emotion/react'
import { colors } from '../_utils/colors'
import { fontPrimary } from '../_utils/commonStyles'
import { ThemeType, get } from '../_theme'

export const table = css``

export const row = (theme: ThemeType) => css`
    font-family: ${fontPrimary};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    line-height: 22px;
    background: ${get(theme, `SimpleStripTable.row.background`)};
    border-radius: 8px;
    transition: all 0.25s;
    height: 54px;

    &:hover {
        background: ${get(theme, `SimpleStripTable.row.hover.background`)};
    }
`

export const autoHeight = css`
    height: 100%;
`

export const attribute = css`
    padding: 16px 24px;
    color: ${colors.neutral500};
    height: 54px;
    box-sizing: border-box;
`

export const value = (theme: ThemeType) => css`
    display: flex;
    align-items: center;
    padding: 0 24px;
    color: ${get(theme, `SimpleStripTable.value.color`)};
    text-align: right;
    justify-content: flex-end;
    height: 54px;
    box-sizing: border-box;
`

export const border = (theme: ThemeType) => css`
    position: relative;

    &:after {
        content: '';
        display: block;
        position: absolute;
        bottom: 0;
        height: 1px;
        background: ${get(theme, `SimpleStripTable.border.background`)};
    }
`
export const borderLeft = css`
    position: relative;

    &:after {
        left: 7px;
        right: 0;
        width: calc(100% - 7px);
    }
`
export const borderRight = css`
    position: relative;

    &:after {
        left: 0;
        right: 7px;
        width: calc(100% - 7px);
    }
`
