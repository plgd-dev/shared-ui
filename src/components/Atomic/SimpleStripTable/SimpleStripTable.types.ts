import { ReactElement, ReactNode } from 'react'

export type Row = {
    attribute: string
    autoHeight?: boolean
    copyValue?: string | boolean
    dataTestId?: string
    key?: string
    required?: boolean
    value: ReactNode | ReactElement
}

export type Props = {
    className?: string
    headerLeft?: string
    headerRight?: string
    i18n?: {
        copy: string
    }
    id?: string
    leftColSize?: number
    rightColSize?: number
    rows: Row[]
}

export const defaultProps: Partial<Props> = {
    leftColSize: 8,
    rightColSize: 4,
}
