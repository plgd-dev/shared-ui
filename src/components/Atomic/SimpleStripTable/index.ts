export { default } from './SimpleStripTable'
export * from './SimpleStripTable'
