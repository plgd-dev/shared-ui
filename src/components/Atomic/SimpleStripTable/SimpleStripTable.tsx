import { FC } from 'react'
import { Props, defaultProps } from './SimpleStripTable.types'
import * as styles from './SimpleStripTable.styles'
import Row from '../Grid/Row'
import Column from '../Grid/Column'
import { ColumnSizeType } from '../Grid/Column/Column.types'

const SimpleStripTable: FC<Props> = (props) => {
    const { className, id, leftColSize, rightColSize, rows } = { ...defaultProps, ...props }

    return (
        <div className={className} id={id}>
            {rows.map((r, key) => (
                <Row css={styles.row} gutters={false} key={key}>
                    <Column size={leftColSize as ColumnSizeType}>
                        <div css={[styles.attribute, styles.border, styles.borderLeft]} data-test-id={r.dataTestId?.concat('-attribute')}>
                            {r.attribute}
                        </div>
                    </Column>
                    <Column size={rightColSize as ColumnSizeType}>
                        <div css={[styles.value, styles.border, styles.borderRight]} data-test-id={r.dataTestId?.concat('-value')}>
                            {r.value}
                        </div>
                    </Column>
                </Row>
            ))}
        </div>
    )
}

export default SimpleStripTable
