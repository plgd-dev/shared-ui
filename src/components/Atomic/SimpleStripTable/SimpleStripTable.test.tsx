import React from 'react'
import { render } from '@testing-library/react'
import SimpleStripTable from './SimpleStripTable'

describe('<SimpleStripTable>', () => {
    it('render correctly - snapshot', () => {
        const { asFragment } = render(
            <SimpleStripTable
                rows={[
                    { attribute: 'attribute1', value: 'value1' },
                    { attribute: 'attribute2', value: 'value2' },
                    { attribute: 'attribute3', value: 'value3' },
                ]}
            />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})
