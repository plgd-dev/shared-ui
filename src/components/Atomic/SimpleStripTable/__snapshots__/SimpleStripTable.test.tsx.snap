// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleStripTable> render correctly - snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: -16px;
  margin-right: -16px;
  box-sizing: border-box;
  margin-right: 0;
  margin-left: 0;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
  background: #FFFFFF;
  border-radius: 8px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  height: 54px;
}

.emotion-0>* {
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0>* {
  padding-right: 0;
  padding-left: 0;
}

.emotion-0:hover {
  background: #f6f7f9;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
  max-width: 100%;
  position: relative;
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 66.66666666666666%;
}

.emotion-2 {
  padding: 16px 24px;
  color: #81868C;
  height: 54px;
  box-sizing: border-box;
  position: relative;
  position: relative;
}

.emotion-2:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 1px;
  background: #E6E9ED;
}

.emotion-2:after {
  left: 7px;
  right: 0;
  width: calc(100% - 7px);
}

.emotion-3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
  max-width: 100%;
  position: relative;
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 33.33333333333333%;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 24px;
  color: #252626;
  text-align: right;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  height: 54px;
  box-sizing: border-box;
  position: relative;
  position: relative;
}

.emotion-4:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 1px;
  background: #E6E9ED;
}

.emotion-4:after {
  left: 0;
  right: 7px;
  width: calc(100% - 7px);
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        attribute1
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        value1
      </div>
    </div>
  </div>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: -16px;
  margin-right: -16px;
  box-sizing: border-box;
  margin-right: 0;
  margin-left: 0;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
  background: #FFFFFF;
  border-radius: 8px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  height: 54px;
}

.emotion-0>* {
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0>* {
  padding-right: 0;
  padding-left: 0;
}

.emotion-0:hover {
  background: #f6f7f9;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
  max-width: 100%;
  position: relative;
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 66.66666666666666%;
}

.emotion-2 {
  padding: 16px 24px;
  color: #81868C;
  height: 54px;
  box-sizing: border-box;
  position: relative;
  position: relative;
}

.emotion-2:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 1px;
  background: #E6E9ED;
}

.emotion-2:after {
  left: 7px;
  right: 0;
  width: calc(100% - 7px);
}

.emotion-3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
  max-width: 100%;
  position: relative;
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 33.33333333333333%;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 24px;
  color: #252626;
  text-align: right;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  height: 54px;
  box-sizing: border-box;
  position: relative;
  position: relative;
}

.emotion-4:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 1px;
  background: #E6E9ED;
}

.emotion-4:after {
  left: 0;
  right: 7px;
  width: calc(100% - 7px);
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        attribute2
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        value2
      </div>
    </div>
  </div>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-left: -16px;
  margin-right: -16px;
  box-sizing: border-box;
  margin-right: 0;
  margin-left: 0;
  font-family: 'Poppins',sans-serif;
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 22px;
  background: #FFFFFF;
  border-radius: 8px;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  height: 54px;
}

.emotion-0>* {
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-0>* {
  padding-right: 0;
  padding-left: 0;
}

.emotion-0:hover {
  background: #f6f7f9;
}

.emotion-1 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
  max-width: 100%;
  position: relative;
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 66.66666666666666%;
}

.emotion-2 {
  padding: 16px 24px;
  color: #81868C;
  height: 54px;
  box-sizing: border-box;
  position: relative;
  position: relative;
}

.emotion-2:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 1px;
  background: #E6E9ED;
}

.emotion-2:after {
  left: 7px;
  right: 0;
  width: calc(100% - 7px);
}

.emotion-3 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 100%;
  max-width: 100%;
  position: relative;
  box-sizing: border-box;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 33.33333333333333%;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 24px;
  color: #252626;
  text-align: right;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  height: 54px;
  box-sizing: border-box;
  position: relative;
  position: relative;
}

.emotion-4:after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 1px;
  background: #E6E9ED;
}

.emotion-4:after {
  left: 0;
  right: 7px;
  width: calc(100% - 7px);
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        attribute3
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        value3
      </div>
    </div>
  </div>
</DocumentFragment>
`;
